Tests/HStackTests.swift

```diff
@@ -1,174 +1,764 @@
+            spacing: 2,
+            thingsToStack: [
+                VStack(
+                    spacing: 1,
+                    thingsToStack: [
+                        UIView().fixed(
+                            size: .init(
+                                width: 100,
+                                height: 10
+                            )
+                        ),
+                        UIView().fixed(
+                            size: .init(
+                                width: 100,
+                                height: 11
+                            )
+                        )
+                    ],
+                    width: 50
+                ),
+                UIView().fixed(
+                    size: .init(
+                        width: 60,
+                        height: 12
+                    )
+                )
+            ]
+        )
         
         let frames = stack.framesForLayout(200)
-        XCTAssertEqual(frames.count, 3)
-        // view1
-        XCTAssertEqual(frames[0].origin.x, 0)
-        XCTAssertEqual(frames[0].origin.y, 0)
-        XCTAssertEqual(frames[0].size.width, fixedVStackWidth)
-        XCTAssertEqual(frames[0].size.height, size1.height)
-        // view2
-        XCTAssertEqual(frames[1].origin.x, 0)
-        XCTAssertEqual(frames[1].origin.y, size1.height + spacing2)
-        XCTAssertEqual(frames[1].size.width, fixedVStackWidth)
-        XCTAssertEqual(frames[1].size.height, size2.height)
-        // view3
-        XCTAssertEqual(frames[2].origin.x, fixedVStackWidth + spacing)
-        XCTAssertEqual(frames[2].origin.y, 0)
-        XCTAssertEqual(frames[2].size.width, size3.width)
-        XCTAssertEqual(frames[2].size.height, size3.height)
+        
+        XCTAssertEqual(
+            frames,
+            [
+                .init(
+                    origin: .zero,
+                    size: .init(
+                        width: 50,
+                        height: 10
+                    )
+                ),
+                .init(
+                    origin: .init(
+                        x: 0,
+                        y: 11
+                    ),
+                    size: .init(
+                        width: 50,
+                        height: 11
+                    )
+                ),
+                .init(
+                    origin: .init(
+                        x: 52,
+                        y: 0
+                    ),
+                    size: .init(
+                        width: 60,
+                        height: 12
+                    )
+                )
+            ]
+        )
     }
     
-    func test_convenience_init_with_thingsToStack_closure() {
-        let layoutMargins = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
-        let spacing: CGFloat = 2
-        let view1 = UILabel()
-        let view2 = UILabel()
+    func test_framesForLayout_when_distribution_is_fill_should_return_expected_frames() {
```

Comment:
This is a test for when subviews exceed the width in the hstack, could be worth mentioning in the test case because the test names help us scan expected behavior.

Reply:
Thanks Shaun, yeah having this in the test name does make it easier to scan Shaun. I did some changes on both unit tests Shaun `.fill` and `.fillEqually` to make it consistent across both 🙇‍♂️